File: javaspace/Solution17.java
import java.util.*;

// Codility - Brackets
public class Solution17 {
    public static void main(String[] args) {
        String S = "([)()]";
        
        System.out.println(solution(S));
    }

    public static int solution(String S) {
        // write your code in Java SE 11
        Stack<Character> stack = new Stack<>();

        for (char c : S.toCharArray()) {
            if ('(' == c || '{' == c || '[' == c ) {
                stack.push(c);
            } else {
                if (stack.isEmpty()) {
                    return 0;
                }
                if (')' == c && '(' != stack.pop()) {
                    return 0;
                } else if ('}' == c && '{' != stack.pop()) {
                    return 0;
                } else if (']' == c && '[' != stack.pop()) {
                    return 0;
                }
            }
        }
        return stack.isEmpty() ? 1 : 0;
    }
}

File: pythonspace/Solution17.py
# Codility - Nesting
# -> 개발중...
def solution(A):
    # write your code in Python 3.6
    if len(S) < 2:
        return 0

    if S[0] == ")":
        return 0
    
    if S[-1] == "(":
        return 0

    stack = []
    
    for i in range(len(S)):
        stack.append(S[i])
        
        pass
    
S = "(()(())())"
solution(S)